python/main.py

```diff
@@ -96,8 +96,12 @@ def left_click_state():
         if left_click_state() and active_state:
             try:
                 for i in range(len(recoil_patterns[active_weapon])):
-                    win32api.mouse_event(0x0001, int(recoil_patterns[active_weapon][i][0]/data["modifier_value"]), int(recoil_patterns[active_weapon][i][1]/data["modifier_value"]))
-                    time.sleep(recoil_patterns[active_weapon][i][2])
+                    if left_click_state() and active_state:
```

Comment:
@asd666q9 
The **active_state** conditional is not required because the overarching if-statement already handles it. You only need to check on the **left_click_state**.